### PR TITLE
Barrier plot

### PR DESCRIPTION
**Pull request by emelinepl**

### CODE REVIEW

File: mmbench/model.py [hunk 1]
```diff
@@ -102,18 +102,18 @@ def get_mopoe(checkpointfile):
     return experiment.mm_vae
 
 
-def eval_mopoe(models, data, modalities, n_samples=10, verbose=1):
+def eval_mopoe(model, data, modalities, n_samples=10, verbose=1):
```
Comment (AGrigis) [inline mmbench/model.py]: n_samples=1 -> cf doc

File: mmbench/plotting.py [hunk 2]
```diff
@@ -98,6 +103,8 @@ def plot_bar(key, rsa, ax=None, figsize=(5, 2), dpi=300, fontsize=16,
         optionally, display pairwise statistics.
     do_one_sample_stars: bool, default True
         optionally, display sampling statistics.
+    yname: str
```
Comment (AGrigis) [inline mmbench/plotting.py]: , default 'model fit (r)'

File: mmbench/plotting.py [hunk 3]
```diff
@@ -213,3 +220,41 @@ def plot_bar(key, rsa, ax=None, figsize=(5, 2), dpi=300, fontsize=16,
...
+def plot_curve(x_data, mat, ax=None, figsize=(5, 2), dpi=300, fontsize=16,
+               fontweight="bold", title=None):
+    """ Display a list of curve.
+
+    Parameters
+    ----------
+    x_data: list
+        the abscissa of the graph.
+    mat: array (n_curve, n_points)
+        the matrix containing ordinates of each curve.
```
Comment (AGrigis) [inline mmbench/plotting.py]: the matrix containing the coordinates of each curve?

File: mmbench/plotting.py [hunk 4]
```diff
@@ -213,3 +220,41 @@ def plot_bar(key, rsa, ax=None, figsize=(5, 2), dpi=300, fontsize=16,
...
+
+
+def plot_curve(x_data, mat, ax=None, figsize=(5, 2), dpi=300, fontsize=16,
+               fontweight="bold", title=None):
+    """ Display a list of curve.
+
+    Parameters
+    ----------
+    x_data: list
+        the abscissa of the graph.
```
Comment (AGrigis) [inline mmbench/plotting.py]: xticks: the list of xtick locations (https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.xticks.html)

File: mmbench/plotting.py [hunk 5]
```diff
@@ -213,3 +220,41 @@ def plot_bar(key, rsa, ax=None, figsize=(5, 2), dpi=300, fontsize=16,
...
+    mat: array (n_curve, n_points)
+        the matrix containing ordinates of each curve.
+    ax: matplotlib.axes.Axes, default None
+        the axes used to display the plot.
```
Comment (AGrigis) [inline mmbench/plotting.py]: the axis

File: mmbench/workflow/barrier.py [hunk 6]
```diff
@@ -137,21 +141,121 @@ def eval_fn(model, loaders, y_train, y_test, eval_fn=None,
...
 
-    barrier_file = os.path.join(benchdir, f"barrier_interp_{dataset}.npz")
+    mat_display(results_test, dataset, outdir, downstream_name, scale)
+    barrier_file = os.path.join(
+        benchdir, f"barrier_interp_{dataset}_{downstream_name}.npz")
     np.savez_compressed(barrier_file, **results_test)
     print_result(f"barrier interpolation: {barrier_file}")
+
+
+def barrier_display(coeffs, l_metrics, model_name, downstream, dataset, outdir,
```
Comment (AGrigis) [inline mmbench/workflow/barrier.py]: move the display functions in plotting?

File: mmbench/workflow/barrier.py [hunk 7]
```diff
@@ -137,21 +141,121 @@ def eval_fn(model, loaders, y_train, y_test, eval_fn=None,
...
+                    scale, sname):
+    """ Save barrier curves for a model
+
+    Parameters
+    ----------
+    coeffs : list
```
Comment (AGrigis) [inline mmbench/workflow/barrier.py]: please remove space before :

File: mmbench/workflow/barrier.py [hunk 8]
```diff
@@ -137,21 +141,121 @@ def eval_fn(model, loaders, y_train, y_test, eval_fn=None,
...
             state2 = model2.state_dict()
             coeffs, metrics = eval_interpolation(
-                model1, state1, state2, [data_train, data_test], eval_fn,
-                n_coeffs=10, eval_kwargs=kwargs)
-            print(coeffs)
-            print(metrics)
-            mat[i1, i2] = np.trapz(metrics, coeffs)
+                copy.deepcopy(model1), state1, state2, [data_train, data_test],
+                eval_fn, n_coeffs=n_coeffs, eval_kwargs=kwargs)
+            ref = metrics[-1] * coeffs + (1 - coeffs) * metrics[0]
```
Comment (AGrigis) [inline mmbench/workflow/barrier.py]: Create a function with documentation.
slope = (y[-1] - y[0]) / (x[-1] - x[0])
intercept = y[0] - slope * x[0]
ref = slope * x + intercept
upref  = [max(y1, y2) for y1, y2 in zip(ref, y)]
downref = [min(y1, y2) for y1, y2 in zip(ref, y)]
A = A(x, upref) - A(x, downref)
